### PR TITLE
Accrue Hard module rewards in multiple coin denoms

### PR DESCRIPTION
**Pull request by denalimarsh**
- [x] Implement multi reward types and update module params
- [x] Update keeper methods to init/accumulate/sync multiple reward denoms
- [x] Update tests so they compile (actual test implementation to be done in follow-on PR)

### CODE REVIEW

File: x/incentive/keeper/rewards.go [hunk 1]
```diff
@@ -319,52 +367,68 @@ func (k Keeper) SynchronizeHardBorrowReward(ctx sdk.Context, borrow hardtypes.Bo
...
-			continue
-		}
-		claim.BorrowRewardIndexes[borrowIndex].RewardFactor = borrowFactor
+		for _, globalRewardIndex := range globalRewardIndexes {
+			userRewardIndex, foundUserRewardIndex := userRewardIndexes.RewardIndexes.GetRewardIndex(globalRewardIndex.CollateralType)
+			if !foundUserRewardIndex {
+				continue
+			}
+			userRewardIndexIndex, foundUserRewardIndexIndex := userRewardIndexes.RewardIndexes.GetFactorIndex(globalRewardIndex.CollateralType)
+			if foundUserRewardIndexIndex {
```
Comment (karzak) [inline x/incentive/keeper/rewards.go:386]:
```suggestion
			if !foundUserRewardIndexIndex {
```

File: x/incentive/keeper/rewards.go [hunk 2]
```diff
@@ -319,52 +367,68 @@ func (k Keeper) SynchronizeHardBorrowReward(ctx sdk.Context, borrow hardtypes.Bo
...
+			}
+			newRewardsAmount := rewardsAccumulatedFactor.Mul(borrow.Amount.AmountOf(coin.Denom).ToDec()).RoundInt()
+			if newRewardsAmount.IsZero() || newRewardsAmount.IsNegative() {
+				continue
+			}
 
-		newRewardsCoin := sdk.NewCoin(types.HardLiquidityRewardDenom, newRewardsAmount)
-		claim.Reward = claim.Reward.Add(newRewardsCoin)
+			factorIndex, foundFactorIndex := userRewardIndexes.RewardIndexes.GetFactorIndex(globalRewardIndex.CollateralType)
+			if foundFactorIndex {
```
Comment (karzak) [inline x/incentive/keeper/rewards.go:403]:
```suggestion
			if !foundFactorIndex {
```

File: x/incentive/keeper/rewards.go [hunk 3]
```diff
@@ -251,33 +291,48 @@ func (k Keeper) SynchronizeHardSupplyReward(ctx sdk.Context, deposit hardtypes.D
...
-			continue
-		}
-		claim.SupplyRewardIndexes[supplyIndex].RewardFactor = supplyFactor
+		for _, globalRewardIndex := range globalRewardIndexes {
+			userRewardIndex, foundUserRewardIndex := userRewardIndexes.RewardIndexes.GetRewardIndex(globalRewardIndex.CollateralType)
+			if !foundUserRewardIndex {
+				continue
+			}
+			userRewardIndexIndex, foundUserRewardIndexIndex := userRewardIndexes.RewardIndexes.GetFactorIndex(globalRewardIndex.CollateralType)
+			if foundUserRewardIndexIndex {
```
Comment (karzak) [inline x/incentive/keeper/rewards.go:310]:
```suggestion
			if !foundUserRewardIndexIndex {
```

File: x/incentive/keeper/rewards.go [hunk 4]
```diff
@@ -251,33 +291,48 @@ func (k Keeper) SynchronizeHardSupplyReward(ctx sdk.Context, deposit hardtypes.D
...
+			}
+			newRewardsAmount := rewardsAccumulatedFactor.Mul(deposit.Amount.AmountOf(coin.Denom).ToDec()).RoundInt()
+			if newRewardsAmount.IsZero() || newRewardsAmount.IsNegative() {
+				continue
+			}
 
-		newRewardsCoin := sdk.NewCoin(types.HardLiquidityRewardDenom, newRewardsAmount)
-		claim.Reward = claim.Reward.Add(newRewardsCoin)
+			factorIndex, foundFactorIndex := userRewardIndexes.RewardIndexes.GetFactorIndex(globalRewardIndex.CollateralType)
+			if foundFactorIndex {
```
Comment (karzak) [inline x/incentive/keeper/rewards.go:327]:
```suggestion
			if !foundFactorIndex {
```

**Review by karzak — CHANGES_REQUESTED** [review]
Looks like everything working once those `!` were added - I managed to get a 'Accumulate' and 'Synchronize' test passing in `rewards_test.go`, so I think we're good to go on to next PR.

File: x/incentive/keeper/rewards.go (hunk 4 above)
Comment (denalimarsh) [reply x/incentive/keeper/rewards.go]: https://github.com/Kava-Labs/kava/pull/785/commits/a949c221d2020e8b4d8a366b01749dd403bf1f2c

File: x/incentive/keeper/rewards.go (hunk 3 above)
Comment (denalimarsh) [reply x/incentive/keeper/rewards.go]: https://github.com/Kava-Labs/kava/pull/785/commits/a949c221d2020e8b4d8a366b01749dd403bf1f2c

File: x/incentive/keeper/rewards.go (hunk 2 above)
Comment (denalimarsh) [reply x/incentive/keeper/rewards.go]: https://github.com/Kava-Labs/kava/pull/785/commits/a949c221d2020e8b4d8a366b01749dd403bf1f2c

File: x/incentive/keeper/rewards.go (hunk 1 above)
Comment (denalimarsh) [reply x/incentive/keeper/rewards.go]: https://github.com/Kava-Labs/kava/pull/785/commits/a949c221d2020e8b4d8a366b01749dd403bf1f2c

**Pull request merged**